tests/sk_not_present.rs

```diff
@@ -0,0 +1,72 @@
+use pam_ssh_agent::{authenticate, PrintLog, SSHAgent};
+use signature::Signer;
+use ssh_agent_client_rs::Error as SACError;
+use ssh_key::{PrivateKey, PublicKey, Signature};
+
+use std::collections::HashMap;
+
+struct DummySshAgent {
+    keys: HashMap<PublicKey, PrivateKey>,
+}
+
+// Generated with `ssh-keygen -t ed25519-sk -f test_ed25519_sk -C test_sk@localhost`
+// with a Yubikey 5c
+const PRIVATE_SK_KEY: &str = include_str!("data/test_ed25519_sk");
+const PUBLIC_SK_KEY: &str = include_str!("data/test_ed25519_sk.pub");
+
+const PRIVATE_KEY: &str = include_str!("data/id_ed25519");
+const PUBLIC_KEY: &str = include_str!("data/id_ed25519.pub");
+
+impl DummySshAgent {
+    fn new() -> DummySshAgent {
+        let sk_key = PrivateKey::from_openssh(PRIVATE_SK_KEY).expect("Failed to parse sk key");
+        let sk_pubkey = PublicKey::from_openssh(PUBLIC_SK_KEY).expect("Failed to parse sk pubkey");
+        let key = PrivateKey::from_openssh(PRIVATE_KEY).expect("Failed to parse test key");
+        let pubkey = PublicKey::from_openssh(PUBLIC_KEY).expect("Failed to parse test pubkey");
+
+        DummySshAgent {
+            keys: HashMap::from([
+                (pubkey, key),
+                (sk_pubkey, sk_key),
+            ]),
+        }
+    }
+}
+
+impl SSHAgent for DummySshAgent {
+    fn list_identities(&mut self) -> ssh_agent_client_rs::Result<Vec<PublicKey>> {
+        let pubkeys: Vec<PublicKey> = [
+            PublicKey::from_openssh(PUBLIC_SK_KEY).expect("Failed to parse sk pubkey"),
+            PublicKey::from_openssh(PUBLIC_KEY).expect("Failed to parse test pubkey"),
+        ].to_vec();
+
+        Ok(pubkeys)
+    }
+
+    fn sign(&mut self, pubkey: &PublicKey, data: &[u8]) -> ssh_agent_client_rs::Result<Signature> {
+        if let Some(key) = self.keys.get(pubkey) {
```

Comment:
This feels like an overly complex way of implementing this. Let's keep it a bit simpler, having the DummySshAgent hold jus a single private key, for the signature that we actually want to create and have sign() be implemented something along the lines of 
```
        if pubkey.algorithm() == Algorithm::SkEd25519 {
            return Err(SACError::RemoteFailure)
        }
        Ok(self.key.key_data().sign(data.as_ref()))
```

This has the additional benefit of not needing to add the "private key" (which is in reality just a reference to the private part of the keypair that lives on your yubikey).